cognite_toolkit/_cdf_tk/templates/_templates.py

```diff
@@ -103,186 +104,280 @@ def build_config(
     return source_by_build_path
 
 
-# TODO: Refactor this function into validation parts for each 'resource_type':
-def check_yaml_semantics(parsed: dict | list, filepath_src: Path, filepath_build: Path, verbose: bool = False) -> bool:
-    """Check the yaml file for semantic errors
+class Resource(Enum):
+    AUTH = "auth"
+    DATA_SETS = "data_sets"
+    DATA_MODELS = "data_models"
+    RAW = "raw"
+    WORKFLOWS = "workflows"
+    TRANSFORMATIONS = "transformations"
+    EXTRACTION_PIPELINES = "extraction_pipelines"
+    TIMESERIES = "timeseries"
+    FILES = "files"
+    FUNCTIONS = "functions"
+    OTHER = "other"
+
+    @classmethod
+    def _missing_(cls, value: object) -> Resource:
+        """Returns Resource.OTHER for any other value."""
+        assert isinstance(value, str)
+        return cls.OTHER
+
+
+def _extract_ext_id_dm_spaces(parsed: dict | list, filepath_src: Path) -> tuple[str | None, str]:
+    if isinstance(parsed, list):
+        raise ToolkitYAMLFormatError(f"Multiple spaces in one file {filepath_src} is not supported.")
```

Comment:
Actually we should support this, the rest of the code is written and tested for it. So this seems to be an old assumption. 

Nice to get a cleanup so we see this, because we need to remove this asap and support list of all resources. Ref @ronpal